packages/synapse-react-client/src/components/TimelinePlot/TimelinePhase.tsx

```diff
@@ -0,0 +1,219 @@
+    },
+  )
+}
+
+const getTimelineData = (timepointData: TimepointData, rowData: Row[]) => {
+  const data = timepointData.timepoints.map((timepoint, index) => {
+    const utcFormattedTimepoint = timepoint.format()
+    const isHoveredOver = index == timepointData.hoverOverIndex
+    const rowId = rowData[index].rowId
+    return {
+      x: [utcFormattedTimepoint, utcFormattedTimepoint, utcFormattedTimepoint],
+      y: [0, 0.5, 1],
+      mode: 'lines',
+      line: {
+        color: isHoveredOver ? 'black' : 'gray',
+        width: 2,
+      },
+      // Add event into in the customdata
+      customdata: [rowId, rowId, rowId],
+      // but tell Plotly that we do not want it to show a hover tooltip (we're going to handle this)
+      hoverinfo: 'none',
+    }
+  })
+  return data
+}
+
+const getLayout = (
+  start: dayjs.Dayjs,
+  end: dayjs.Dayjs,
+  color: string,
+  annotateTime?: number,
+  annotateTimeUnits?: ManipulateType,
+): Partial<Layout> => {
+  const annotations: Partial<Plotly.Annotations>[] = [
+    {
+      x: start.add(1, 'day').format(),
+      y: -1,
+      text: '                             ', // hacky annotation in the middle of the plot so it does not shift when showing other annotations
+      showarrow: false,
+      textangle: '270',
+      height: 15,
+    },
+  ]
+  if (annotateTime && annotateTimeUnits) {
+    const x = start.add(annotateTime, annotateTimeUnits)
+    annotations.push({
+      x: x.format(),
+      y: -1,
+      text: `${annotateTime} ${pluralize(annotateTimeUnits, annotateTime)}`,
+      showarrow: false,
+      textangle: '270',
+      height: 15,
+    })
+  }
+  return {
+    hovermode: 'closest',
+    dragmode: false, //disallow interaction
+    showlegend: false,
+    xaxis: {
+      showgrid: false,
+      showticklabels: false,
+      showline: false,
+      zeroline: false,
+    },
+
+    yaxis: {
+      range: [-1, 1],
+      showgrid: false,
+      zeroline: false,
+      showline: false,
+      showticklabels: false,
+    },
+    annotations: annotations,
+    // Each phase has a shape
+    shapes: [
+      {
+        type: 'rect',
+        x0: start.subtract(end.diff(start) / 10).format(), // add 10% to the left side of the graph
+        x1: end.add(end.diff(start) / 3).format(), // add 33% to right side of the graph
+        y0: 0.25,
+        y1: 0.75,
+        fillcolor: color,
+        opacity: 0.8,
+        line: {
+          width: 0,
+        },
+      },
+    ],
+    margin: {
+      l: 0,
+      r: 0,
+      t: 60,
+      b: 60,
+    },
+    // autosize: false,
+  }
+}
+
+type TimelinePhaseProps = {
+  name: string
+  color: string
+  rowData: Row[]
+  schema: ObservationCardSchema
+  widthPx: number
+}
+
+const TimelinePhase = ({
+  color,
+  rowData,
+  schema,
+  widthPx,
+}: TimelinePhaseProps) => {
+  const [clickEvent, setClickEvent] = useState<Plotly.PlotMouseEvent>()
+  const [hoverEvent, setHoverEvent] = useState<Plotly.PlotHoverEvent>()
+  const start = dayjs()
```

Comment:
The plot x axis are dates.  The observation data are all durations, so we just add the duration to the current date to find where each point should be on the x-axis.